src/actions/PrecursorAction.C

```diff
@@ -233,7 +244,38 @@ PrecursorAction::act()
         std::string kernel_name = "VelocityFunctionOutflowBC_" + var_name + "_" + _object_suffix;
         _problem->addBoundaryCondition("VelocityFunctionOutflowBC", kernel_name, params);
       }
+      // INFLOW
+      if (getParam<bool>("loop_precs"))
+      {
+          // this SHOULD work for both constant and nonconstant flows as long as
+          // nonconstant flows implemented in the Controls module by
+          // setting values called uu, vv, ww.
+          if (!getParam<bool>("constant_velocity_values"))
+              mooseError("Variable, looped precursor advection requires that variable"
+                         "velocity has the values uu, vv, ww set through the controls"
+                         "module, NOT simply specifying functions through the"
+                         "precursors block.");
+          InputParameters params = _factory.getValidParams("PostprocessorInflowBC");
+          params.set<NonlinearVariableName>("variable") = var_name;
+          params.set<std::vector<BoundaryName>>("boundary") =
+              getParam<std::vector<BoundaryName>>("inlet_boundaries");
+          params.set<Real>("uu") = getParam<Real>("u_def");
+          params.set<Real>("vv") = getParam<Real>("v_def");
+          params.set<Real>("ww") = getParam<Real>("w_def");
+          params.set<PostprocessorName>("postprocessor") = 
+              "Inlet_SideAverageValue_"+
+              var_name + "_" + _object_suffix;
+
+          // OK, Alex called it kernel_name despite it being a BC,
+          // and it doesn't matter, but let's follow suit.
+          std::string kernel_name = "PostprocessorInflowBC_" + var_name + 
+              "_" + _object_suffix;
+          _problem->addBoundaryCondition("PostprocessorInflowBC", kernel_name, params);
+      }
+
+      // fixed concentration input: no use at the moment so commented out
```

Comment:
Best practice is to delete unused code, rather than commenting it out. You can always get old code back if it's in git!.

Reply:
Oooh good point. Changed this.